scripts/pre-commit.sh

```diff
@@ -75,12 +83,24 @@ cd "$base_dir" || {
 
 rc=0
 
-echo "Building federation binaries"
-build-binaries
-
 echo "Downloading test dependencies"
 ./scripts/download-binaries.sh
 
+echo "Checking that 'kubebuilder generate' is up-to-date"
+if output=$(git status --porcelain) && [ -z "$output" ]; then
```

Comment:
Can we instead encapsulate this logic in a function e.g. `run-kubebuilder-generate`?

Reply:
Done